internal/app/summon/command/temp_factory_test.go

```diff
@@ -0,0 +1,88 @@
+package command
+
+import (
+	"io/ioutil"
+	"os"
+	"strings"
+	"testing"
+
+	. "github.com/smartystreets/goconvey/convey"
+)
+
+// TODO: Test TempFactory.Cleanup()
+// TODO: Test TempFactory.Push()
+
+func splitEq(s string) (string, string) {
+	a := strings.SplitN(s, "=", 2)
+	return a[0], a[1]
+}
+
+type envSnapshot struct {
+	env []string
+}
+
+func ClearEnv() *envSnapshot {
+	e := os.Environ()
+
+	for _, s := range e {
+		k, _ := splitEq(s)
+		os.Setenv(k, "")
+	}
+	return &envSnapshot{env: e}
+}
+
+func (e *envSnapshot) RestoreEnv() {
+	ClearEnv()
+	for _, s := range e.env {
+		k, v := splitEq(s)
+		os.Setenv(k, v)
+	}
+}
+
+func TestTempFactory_NewTempFactory(t *testing.T) {
+	Convey("Uses constructor arg path if provided", t, func() {
+		tempFactory := NewTempFactory("somedir")
+
+		So(tempFactory, ShouldResemble, TempFactory{
+			files: []string(nil),
+			path:  "somedir",
+		})
+	})
+
+	Convey("When constructor path is not provided", t, func() {
+		env := ClearEnv()
+		defer env.RestoreEnv()
+
+		Convey("tries using shared memory path first", func() {
+			tempFactory := NewTempFactory("")
+
+			So(tempFactory, ShouldResemble, TempFactory{
+				files: []string(nil),
+				path:  "/dev/shm",
```

Comment:
This goes against my comment above to make `sharedMemoryDir` private, but if we are going to write this test, then the `path` value should def not be hardcoded.  Instead it shoud be `path: TempFactory.SharedMemoryDir`.  

*However*, I'm wondering what we're really testing here?  This feels like a test of impl detail.  Got me thinking about what the purpose of this type really is.  Then I looked at the  definition and noticed the summon ext, and realized that it's not really clear.  So I'd like to know what you think about that.

Reply:
Either way, you have to somewhere test the exact value of the variable. Either you:
- Expose `TempFactory.SharedMemoryDir`
- Test that `TempFactory.path` == `TempFactory.SharedMemoryDir`
- Test that `TempFactory.SharedMemoryDir` == `/dev/shm`
or you 
- Don't care about where the path value comes from as long as it's the expected value

In both cases, you have to assert in the tests that the returned shared memory dir matches the one you expect and that can easily be done by string cmp since the value is a static const.

Since I realize now that we don't need the `TempFactory.SharedMemoryDir` public as nothing is using that outside of the class, I'll swap it to private.

Reply:
> Since I realize now that we don't need the TempFactory.SharedMemoryDir public as nothing is using that outside of the class, I'll swap it to private.

Reply:
> what is the behavior of this class, really? 

Looks to me to be in charge of creation and cleanup of all tempfiles created by it. It uses some logic to try to find a secure-ish place for created files.

> If you don't want to make that big a change at this point, that's ok with me.

Bigger change will touch other parts of the system and would open it up to more review changes and manual test invalidation for originally a very small change into the codebase which goes counter to the Git workflow in my opinion. If we need something specific fixed we should open new issues up.

> Test that TempFactory.path == TempFactory.SharedMemoryDir

I think that it doesn't need to be public but it would be good if we could use some Golang built-in to ensure that it's the right value. Hardcoded value like we use here is not really portable nor flexible.